### PR TITLE
Fix reporting of zoom in the MIDI editor when there is more than one take

### PR DESCRIPTION
**Pull request by RDMurray**

### CODE REVIEW

**Comment by ScottChesworth** [comment]
All seems good on the front-end man. Is this ready to be merged?

**Comment by RDMurray** [comment]
Not quite, @jcsteh should probably take a look at this first. I just committed a change that drastically improves performance for large midi items.

Feel free to test this with huge midi items that have multiple takes and over 100000 events. See if you can make it crash. :)

**Comment by ScottChesworth** [comment]
Just done a test where I had 7 takes, each one being a randomly picked bunch of MIDI files glued together. The lightest take had just over 100000 events, the biggest had 277389. Changing the zoom worked fine. It was noticeably more responsive than navigating between notes lol.

**Comment by Lo-lo78** [comment]
> Not quite, @jcsteh should probably take a look at this first. I just committed a change that drastically improves performance for large midi items.
> 
> Feel free to test this with huge midi items that have multiple takes and over 100000 events. See if you can make it crash. :)

I tried with items of 200000 midi events and the problem is that it slows everything down.
I think it depends on NVDA.
If NVDA receives too many events it also slows down Reaper.
I think nothing can be done.
It's the same problem with the Surge XT.
Juce is accessible but Surge has too many events that it sends to NVDA and is therefore a little slow to open the plugin.
My guess is that NVDA has some issues with very large lists and with applications/plugins that have a lot of events it also slows down the application/plugin.
I hope I'm wrong!

File: src/midiEditorCommands.cpp [hunk 1]
```diff
@@ -50,27 +51,33 @@ int getItemPPQ(MediaItem* item) {
 	return ppq;
 }
 
-// return the midi editor zoom ratio of the item 
-double getMidiZoomRatio(MediaItem* item) {
+struct FreeReaperPtr {
+	void operator()(void* p)
+	{
```
Comment (jcsteh) [inline src/midiEditorCommands.cpp:56]:
```suggestion
	void operator()(void* p) {
```

**Review by jcsteh — APPROVED** [review]
Ah, this is much nicer. Thank you.

**Pull request merged**